frontend/src/features/admin-form/responses/ResponsesPage/storage/UnlockedResponses/DownloadButton.tsx

```diff
@@ -38,35 +71,72 @@ export const DownloadButton = (): JSX.Element => {
     })
   }, [downloadParams, handleExportCsvMutation])
 
+  const resetDownload = useCallback(() => {
+    setDownloadCount(0)
+    setProgressModalTimeout(null)
+    abortDecryption()
+  }, [abortDecryption])
+
+  const handleAbortDecryption = useCallback(() => {
+    resetDownload()
+    onProgressModalClose()
+  }, [onProgressModalClose, resetDownload])
+
   return (
-    <Box gridArea="export" justifySelf="flex-end">
-      <Menu placement="bottom-end">
-        {({ isOpen }) => (
-          <>
-            <MenuButton
-              as={Button}
-              isDisabled={!downloadParams}
-              isLoading={handleExportCsvMutation.isLoading}
-              isActive={isOpen}
-              aria-label="Download options"
-              rightIcon={isOpen ? <BxsChevronUp /> : <BxsChevronDown />}
-            >
-              Download
-            </MenuButton>
-            <Menu.List>
-              <Menu.Item onClick={handleExportCsvNoAttachments}>
-                CSV only
-              </Menu.Item>
-              <Menu.Item onClick={handleExportCsvWithAttachments}>
-                CSV with attachments
-                <Badge ml="0.5rem" colorScheme="success">
-                  beta
-                </Badge>
-              </Menu.Item>
-            </Menu.List>
-          </>
-        )}
-      </Menu>
-    </Box>
+    <>
+      {responsesCount && (
+        <DownloadWithAttachmentModal
+          responsesCount={responsesCount}
```

Comment:
nit: feel like we're passing this prop everywhere when we can just use `useStorageResponsesContext`

Reply:
wanted to keep the modal "pure" for storybook, which i have now added stories for hehe